cgosqlite/cgosqlite.go

```diff
@@ -148,8 +151,17 @@ func (stmt *Stmt) ClearBindings() error {
 	return errCode(C.sqlite3_clear_bindings(stmt.stmt))
 }
 
-func (stmt *Stmt) ResetAndClear() error {
-	return errCode(C.reset_and_clear(stmt.stmt))
+func (stmt *Stmt) ResetAndClear() (time.Duration, error) {
+	if stmt.start != (C.struct_timespec{}) {
+		var d C.int64_t
+		err := errCode(C.reset_and_clear(stmt.stmt, &stmt.start, &d))
```

Comment:
I think this'll force d onto the stack. Do you have any benchmarks and/or alloc tests that cover this? You may want to return a struct from reset_and_clear instead. (cgo is so awkward.)

Reply:
Fixed by cheating, I put some scratch variables inside the *Stmt object that is already heap allocated. (Use of it is single-threaded.)